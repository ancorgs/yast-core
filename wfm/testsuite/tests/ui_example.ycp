{
   //

   define DisplayNoSuccessWarning1() ``{
	UI(``{
	string message = _("
The test of the currently selected video settings
has not been passed successfully.

You may now try other video settings or you may
skip the X11 configuration altogether if the test
fails again and again with different settings.
");

	    OpenDialog(
		       `opt(`decorated),
		       `HBox(
			     `HSpacing(1),
			     `VBox(
				   `VSpacing(0.2),
				   `Label( message ),
				   // check box: User can decide if he wants to see this warning again
				   `CheckBox(`id(`box), _("Don't tell me again."), false),
				   `VSpacing(0.2),
				   // ok pushbutton: confirm the dialog
				   `PushButton(`opt(`default), _("&OK")),
				   `VSpacing(0.2)
				   ),
			     `HSpacing(1)
			     )
		       );
	    UserInput();

	boolean dont_tell_me = UI(`QueryWidget(`id(`box), `Value ));

	CloseDialog();

	return( ! dont_tell_me );	// invert
        });
    }

	UI(``{
    global define DisplayNoSuccessWarning2()
	``{
	    string message = _("
The test of the currently selected video settings
has not been passed successfully.

You may now try other video settings or you may
skip the X11 configuration altogether if the test
fails again and again with different settings.
");

	    OpenDialog(
		       `opt(`decorated),
		       `HBox(
			     `HSpacing(1),
			     `VBox(
				   `VSpacing(0.2),
				   `Label( message ),
				   // check box: User can decide if he wants to see this warning again
				   `CheckBox(`id(`box), _("Don't tell me again."), false),
				   `VSpacing(0.2),
				   // ok pushbutton: confirm the dialog
				   `PushButton(`opt(`default), _("&OK")),
				   `VSpacing(0.2)
				   ),
			     `HSpacing(1)
			     )
		       );
	    UserInput();

	    boolean dont_tell_me = QueryWidget(`id(`box), `Value );

	    CloseDialog();

	    return( ! dont_tell_me );	// invert
    }
	});

   define DisplayNoSuccessWarning3() ``{
	string message = _("
The test of the currently selected video settings
has not been passed successfully.

You may now try other video settings or you may
skip the X11 configuration altogether if the test
fails again and again with different settings.
");

	UI(``{
	    OpenDialog(
		       `opt(`decorated),
		       `HBox(
			     `HSpacing(1),
			     `VBox(
				   `VSpacing(0.2),
				   `Label( message ),
				   // check box: User can decide if he wants to see this warning again
				   `CheckBox(`id(`box), _("Don't tell me again."), false),
				   `VSpacing(0.2),
				   // ok pushbutton: confirm the dialog
				   `PushButton(`opt(`default), _("&OK")),
				   `VSpacing(0.2)
				   ),
			     `HSpacing(1)
			     )
		       );
	    UserInput();

	boolean dont_tell_me = UI(`QueryWidget(`id(`box), `Value ));

	CloseDialog();

	return( ! dont_tell_me );	// invert
        });
    }

    boolean show_no_success_warning1 = DisplayNoSuccessWarning1();
    boolean show_no_success_warning2 = UI::DisplayNoSuccessWarning2();
    boolean show_no_success_warning3 = DisplayNoSuccessWarning3();

}
