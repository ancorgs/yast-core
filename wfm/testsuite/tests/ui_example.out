({
  global define DisplayNoSuccessWarning2() ``{
  string `message = _("\nThe test of the currently selected video settings\nhas not been passed successfully.\n\nYou may now try other video settings or you may\nskip the X11 configuration altogether if the test\nfails again and again with different settings.\n");
  OpenDialog (`opt (`decorated), `HBox (`HSpacing (1), `VBox (`VSpacing (0.200000000000000), `Label (message), `CheckBox (`id (`box), _("Don't tell me again."), false), `VSpacing (0.200000000000000), `PushButton (`opt (`default), _("&OK")), `VSpacing (0.200000000000000)), `HSpacing (1)));
  UserInput ();
  boolean `dont_tell_me = QueryWidget (`id (`box), `Value);
  CloseDialog ();
  return ! (dont_tell_me);
}
;
}
)
({
  string `message = _("\nThe test of the currently selected video settings\nhas not been passed successfully.\n\nYou may now try other video settings or you may\nskip the X11 configuration altogether if the test\nfails again and again with different settings.\n");
  OpenDialog (`opt (`decorated), `HBox (`HSpacing (1), `VBox (`VSpacing (0.200000000000000), `Label (message), `CheckBox (`id (`box), _("Don't tell me again."), false), `VSpacing (0.200000000000000), `PushButton (`opt (`default), _("&OK")), `VSpacing (0.200000000000000)), `HSpacing (1)));
  UserInput ();
  boolean `dont_tell_me = UI::`QueryWidget (`id (`box), `Value);
  CloseDialog ();
  return ! (dont_tell_me);
}
)
(DisplayNoSuccessWarning2 ())
({
  OpenDialog (`opt (`decorated), `HBox (`HSpacing (1), `VBox (`VSpacing (0.200000000000000), `Label (message), `CheckBox (`id (`box), _("Don't tell me again."), false), `VSpacing (0.200000000000000), `PushButton (`opt (`default), _("&OK")), `VSpacing (0.200000000000000)), `HSpacing (1)));
  UserInput ();
  boolean `dont_tell_me = UI::`QueryWidget (`id (`box), `Value);
  CloseDialog ();
  return ! (dont_tell_me);
}
)
(nil)
